src/adis16448/adis16448.cxx

```diff
@@ -55,24 +56,81 @@ using namespace upm;
 ////////////////////////////////////////////////////////////////////////////
 ADIS16448::ADIS16448(int bus, int rst)
 {
-// Configure I/O
-        //Initialize RST pin
-        if ( !(_rst = mraa_gpio_init(rst)) ) 
-          {
-            throw std::invalid_argument(std::string(__FUNCTION__) +
-                                        ": mraa_gpio_init() failed, invalid pin?");
-            return;
-          }
+  // Configure I/O
+  //Initialize RST pin
+  if ( !(_rst = mraa_gpio_init(rst)) )
+    {
+      throw std::invalid_argument(std::string(__FUNCTION__) +
+                                  ": mraa_gpio_init() failed, invalid pin?");
+      return;
+    }
 	mraa_gpio_dir(_rst, MRAA_GPIO_IN); //Set direction as INPUT
 
-        // Configure SPI
-        if ( !(_spi = mraa_spi_init(bus)) ) 
-          {
-            throw std::invalid_argument(std::string(__FUNCTION__) +
-                                        ": mraa_spi_init() failed");
-            return;
-          }
-        configSPI();
+  // Configure SPI
+  if ( !(_spi = mraa_spi_init(bus)) )
+    {
+      throw std::invalid_argument(std::string(__FUNCTION__) +
+                                  ": mraa_spi_init() failed");
+      return;
+    }
+  configSPI();
+}
+
+ADIS16448::ADIS16448(std::string initStr) : mraaIo(initStr)
+{
+  mraa_io_descriptor* descs = mraaIo.getMraaDescriptors();
+
+  std::vector<std::string> upmTokens;
+
+  if (!mraaIo.getLeftoverStr().empty()) {
+      upmTokens = UpmStringParser::parse(mraaIo.getLeftoverStr());
+  }
+
+  // Configure I/O
+  //Initialize RST pin
+  if(!descs->gpios)
+  {
+    throw std::invalid_argument(std::string(__FUNCTION__) +
+                                  ": mraa_gpio_init() failed, invalid pin?");
+  }
+  else
+  {
+    if( !(_rst = descs->gpios[0]) )
+    {
+      throw std::invalid_argument(std::string(__FUNCTION__) +
+                              ": mraa_gpio_init() failed, invalid pin?");
+
+    }
+  }
+  mraa_gpio_dir(_rst, MRAA_GPIO_IN); //Set direction as INPUT
+
+  // Configure SPI
+  if(!descs->spis)
+  {
+    throw std::invalid_argument(std::string(__FUNCTION__) +
```

Comment:
Make sure that the log message is explicit enough for the user: it comes from an spi port, what actual port was being targeted.